lib/tape.js

```diff
@@ -0,0 +1,259 @@
+		callee: {name: 'require'},
+		arguments: arg => arg[0].value === 'tape'
+	})
+	.filter(p => p.value.arguments.length === 1)
+	.forEach(p => {
+		p.node.arguments[0].value = 'ava';
+		testFunctionName = p.parentPath.value.id.name;
+	});
+
+	ast.find(j.ImportDeclaration, {
+		source: {
+			type: 'Literal',
+			value: 'tape'
+		}
+	})
+	.forEach(p => {
+		p.node.source.value = 'ava';
+		testFunctionName = p.value.specifiers[0].local.name;
+	});
+
+	return testFunctionName;
+}
+
+export default function tapeToAva(fileInfo, api) {
+	const j = api.jscodeshift;
+	const ast = j(fileInfo.source);
+
+	const testFunctionName = updateTapeRequireAndImport(j, ast);
+
+	if (testFunctionName) {
+		const transforms = [
+			function detectUnsupportedNaming() {
+				// Currently we only support "t" as the test argument name
+				const validateTestArgument = p => {
+					const lastArg = p.value.arguments[p.value.arguments.length - 1];
+					if (lastArg && lastArg.params && lastArg.params[0]) {
+						const lastArgName = lastArg.params[0].name;
+						if (lastArgName !== 't') {
+							throw new Error(`Argument to test function should be named "t" not "${lastArgName}"`);
+						}
+					}
+				};
+
+				ast.find(j.CallExpression, {
+					callee: {
+						object: {name: testFunctionName}
+					}
+				})
+				.forEach(validateTestArgument);
+
+				ast.find(j.CallExpression, {
+					callee: {name: testFunctionName}
+				})
+				.forEach(validateTestArgument);
+			},
+
+			function detectUnsupportedFeatures() {
+				ast.find(j.CallExpression, {
+					callee: {
+						object: {name: 't'},
+						property: ({name}) => unsupportedTestFunctions.has(name)
+					}
+				})
+				.forEach(p => {
+					throw new Error(`Codemod transformation of "${p.value.callee.property.name}" is not supported`);
+				});
+
+				ast.find(j.CallExpression, {
+					callee: {
+						object: {name: testFunctionName},
+						property: {name: 'createStream'}
+					}
+				})
+				.forEach(() => {
+					throw new Error('Codemod transformation of "createStream" is not supported');
+				});
+			},
+
+			function updateAssertions() {
+				ast.find(j.CallExpression, {
+					callee: {
+						object: {name: 't'},
+						property: ({name}) => Object.keys(tapeToAvaAsserts).indexOf(name) >= 0
+					}
+				})
+				.forEach(p => {
+					const property = p.node.callee.property;
+					property.name = tapeToAvaAsserts[property.name];
+				});
+			},
+
+			function testOptionArgument() {
+				// Convert Tape option parameters, test([name], [opts], cb)
+				ast.find(j.CallExpression, {
+					callee: {name: testFunctionName}
+				}).forEach(p => {
+					p.value.arguments.forEach(a => {
+						if (a.type === 'ObjectExpression') {
+							a.properties.forEach(tapeOption => {
+								const tapeOptionKey = tapeOption.key.name;
+								const tapeOptionValue = tapeOption.value.value;
+								if (tapeOptionKey === 'skip' && tapeOptionValue === true) {
+									p.value.callee.name += '.cb.serial.skip';
```

Comment:
Neat! I didn't know `recast` would support this. I've always built out a bunch of member expressions.